tests/drivers/timer/nrf_grtc_timer/src/main.c

```diff
@@ -40,23 +40,44 @@ ZTEST(nrf_grtc_timer, test_get_ticks)
 
 	for (uint32_t i = 0; i < NUMBER_OF_TRIES; i++) {
 		/* Absolute timeout 1ms in the past */
-		t = Z_TIMEOUT_TICKS(Z_TICK_ABS(sys_clock_tick_get() - K_MSEC(1).ticks));
-
-		exp_ticks = z_nrf_grtc_timer_read() - K_MSEC(1).ticks * CYC_PER_TICK;
-		ticks = z_nrf_grtc_timer_get_ticks(t);
-		zassert_true((ticks >= (exp_ticks - CYC_PER_TICK + 1)) &&
-				     (ticks <= (exp_ticks + GRTC_SLEW_TICKS)),
-			     "Unexpected result %" PRId64 " (expected: %" PRId64 ")", ticks,
-			     exp_ticks);
+		uint64_t curr_tick = sys_clock_tick_get();
+		uint64_t curr_grtc_tick = z_nrf_grtc_timer_read();
+		uint64_t curr_tick2 = sys_clock_tick_get();
+
+		/* Skip check if system tick changed during reading of system clock
```

Comment:
Skipping this part could lead to false positives. How about retrying until `curr_tick == curr_tick2`?
Unless there are targets where tick during reading always happens.

Reply:
done.